client/src/components/exercises/ExercisesTable/index.js

```diff
@@ -0,0 +1,122 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import useStyle from './style';
+import AddExercise from '../AddExercise';
+import ExerciseItems from '../ExerciseItems';
+import CustomErrorMessage from '../../food/CustomErrorMessage';
+import Loading from '../../Loading';
+
+function ExercisesTable() {
+  const [showLoading, setShowLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const classes = useStyle();
+  const [exercisesList, setExercisesList] = useState([
+    { id: 1, exercise_name: 'bicycling', met: 7.5, caloriesPerHour: 496 },
+  ]);
+  const [open, setOpen] = useState(false);
+  const [duration, setDuration] = useState(0);
+  const [exerciseName, setExerciseName] = useState('running');
+  const [exerciseCalories, setExerciseCalories] = useState(500);
+  const [exerciseId, setExerciseId] = useState(0);
+  const [userExercisesList, setUserExercisesList] = useState([
+    { id: 2, exercises_id: 1, duration: 30 },
+  ]);
+  const [getData, setGetData] = useState(false);
+  const handleClickOpen = (
+    exerciseNameVar,
+    exerciseCaloriesVar,
+    exerciseIdVar
+  ) => {
+    setExerciseName(exerciseNameVar);
+    setExerciseCalories(exerciseCaloriesVar);
+    setExerciseId(exerciseIdVar);
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  const handlePostRequest = async (id, exerciseDuration) => {
+    try {
+      const {
+        data: { data },
+      } = await axios.post(`/api/v1/exercises/${id}`, {
+        exerciseDuration,
+      });
+      setGetData((item) => !item);
+      return console.log(data);
+    } catch (err) {
+      setGetData((item) => !item);
+      return console.log(err.response);
+    }
+  };
+
+  useEffect(async () => {
```

Comment:
This entire effect could be moved into its own file and called here, as a custom hook. That would separate this large file nicely.